midcli/__main__.py

```diff
@@ -77,8 +78,43 @@ def get_message():
 
         return prompt_app
 
+    def _show_banner(self):
+        self._show_urls()
+
+        print('Type "help" to list available commands.')
+        print()
+
+    def _show_urls(self):
+        with self.context.get_client() as c:
+            try:
+                urls = c.call('system.general.get_ui_urls')
+            except Exception:
+                pass
+            else:
+                print()
+                print('The web user interface is at:')
+                for url in urls:
+                    print(url)
+                print()
+
+    def _should_switch_to_shell(self):
+        if is_main_cli():
+            try:
+                with self.context.get_client() as c:
+                    return not c.call('system.advanced.config')['consolemenu']
+            except Exception:
+                pass
+
+        return False
+
     def _read_kmsg(self):
         with open("/dev/kmsg") as f:
+            # Skip existing messages
```

Comment:
Forgive me, I don't understand this change? Before `readline()` would block until new messages were written now I don't understand what this does.

Reply:
Every time Linux spits some messages on tty1 we want to wait 3 seconds after it stops writing and then re-print our CLI prompt. That's why we read from `/dev/kmsg` and react on new messages.

When we open `/dev/kmsg` it already contains some messages that were written to the console before CLI was started. That made CLI start, print prompt, then print prompt again after 3 seconds (as it has seen the kernel messages that were in ring buffer before CLI was started). With this change we skip all the existing contents of `/dev/kmsg` before entering the read loop described above.

Reply:
Thanks!